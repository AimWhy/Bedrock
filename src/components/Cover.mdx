---
name: Cover
menu: Components
---

import { Playground, Props } from 'docz';
import Cover from '@bedrock-layout/cover';

# Cover

## Use case

The Cover is designed to cover a predefinded area, 100vh by default, and vertically center it's child.

## How to install

If you haven't already, install `styled-components`:

```bash
  npm install styled-components
```

Then install the Cover component

```bash
  npm install @bedrock-layout/cover
```

## Basic usage

<Playground>
  <Cover minHeight="50vh" gutter="sm" top={<div>I'm on the Top</div>}>
    <div>
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Fuga consequuntur
      corrupti beatae commodi vitae, perspiciatis totam provident architecto
      doloribus aperiam sapiente, incidunt nihil suscipit voluptatibus tempore
      est dolor! Iusto, vero.
    </div>
  </Cover>
</Playground>

## data-attribute

For purposes of styling, the outer wrapper as the `data-bedrock-layout-cover` data attribute. The `top`, `children`, and `bottom` are wrapped in an elements with a data attributes of `data-bedrock-layout-cover-top`, `data-bedrock-layout-cover-child`, and `data-bedrock-layout-cover-bottom` respectively.

## Storybook

[Checkout storybook](https://storybook.bedrock-layout.dev/?path=/story/cover--basic) for more details.

## Properties

| Property  | Description                       | Type                                                                      | Default |
| --------- | --------------------------------- | ------------------------------------------------------------------------- | ------- |
| minHeight | minHeight of the cover            | any valid css size unit as a string (will revert to default if not valid) | "100vh" |
| gutter    | sets space inbetween each element | `"none" , "xxs" ,"xs" , "sm", "md", "lg", "lg", "xl", "xxl"`              | `lg`    |
| children  | takes a single child              | ReactNode                                                                 | -       |
| top       | takes a single child              | ReactNode                                                                 | -       |
| bottom    | takes a single child              | ReactNode                                                                 | -       |
