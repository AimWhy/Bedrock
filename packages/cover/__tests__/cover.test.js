import React from 'react';
import { create } from 'react-test-renderer';
//import { ThemeProvider } from 'styled-components';
import Cover from '../src';

const Lorem = () => (
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in
    vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque
    sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin
    elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut
    imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique
    senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices,
    quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus
    tortor et enim.
  </p>
);

describe('Cover', () => {
  describe('correct usage', () => {
    test('Cover is not null', () => {
      expect(Cover).toBeTruthy();
    });

    it('renders default width', () => {
      const cover = create(
        <Cover>
          <Lorem />
        </Cover>
      );
      expect(cover.toJSON()).toMatchSnapshot();
    });

    // it('renders custom width', () => {
    //   const cover = create(
    //     <Cover maxWidth={320}>
    //       <Lorem />
    //     </Cover>
    //   );
    //   expect(cover.toJSON()).toMatchSnapshot();
    // });

    // it('renders with theme overrides', () => {
    //   const cover = create(
    //     <ThemeProvider theme={{ breakPoints: { medium: 1600 } }}>
    //       <Cover>
    //         <Lorem />
    //       </Cover>
    //     </ThemeProvider>
    //   );
    //   expect(cover.toJSON()).toMatchSnapshot();
    // });
  });

  // describe('incorrect usage', () => {
  //   let originalError;
  //   let spy;
  //   beforeEach(() => {
  //     originalError = console.error;
  //     spy = jest.fn();
  //     console.error = spy;
  //   });
  //   afterEach(() => {
  //     console.error = originalError;
  //   });

  //   it('renders default with console error with no children', () => {
  //     expect(spy.mock.calls.length).toBe(0);

  //     const errorStack = create(
  //       <Cover maxWidth='incorrect'>
  //         <Lorem />
  //       </Cover>
  //     );

  //     expect(spy.mock.calls.length).toBe(1);
  //     expect(errorStack.toJSON()).toMatchSnapshot();
  //   });
  // });
});
