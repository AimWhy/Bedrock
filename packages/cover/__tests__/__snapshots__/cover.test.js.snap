// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cover correct usage renders default width 1`] = `
.c2 {
  margin-top: auto;
  margin-bottom: auto;
  margin-block-start: auto;
  margin-block-end: auto;
}

.c0 {
  box-sizing: border-box;
  padding: 0.5rem;
  --gutter: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  min-height: 100vh;
}

.c0 > *:not(.c1) {
  margin-top: var(--gutter);
  margin-bottom: var(--gutter);
  margin-block-start: var(--gutter);
  margin-block-end: var(--gutter);
}

.c0 > :first-child:not(.c1) {
  margin-top: 0;
}

.c0 > :last-child:not(.c1) {
  margin-bottom: 0;
}

<div
  className="sc-AxjAm c0"
>
  <div
    className="c1 c2"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nullam in vestibulum tortor, vitae venenatis lectus. Praesent gravida dapibus neque sit amet molestie. Morbi blandit eu dolor a luctus. Vestibulum sollicitudin elit ac nunc scelerisque rhoncus. Nulla felis sapien, condimentum ut imperdiet vel, aliquet id ante. Pellentesque habitant morbi tristique senectus et netus et malesuada fames ac turpis egestas. Quisque ultrices, quam nec scelerisque malesuada, lectus elit semper diam, ac placerat purus tortor et enim.
    </p>
  </div>
</div>
`;
